### DISCUSSION TITLE
Multiple entrypoint executables

### DISCUSSION THREAD

**Discussion by LewisGaul** [pyinstaller/pyinstaller#6634] [Features & Development]
Hey everyone :) I was wondering if there's a supported way to create multiple executables in a single package, sharing the stdlib, 3rd party code and potentially data files (as described [here](https://stackoverflow.com/questions/66258907/pyinstaller-getting-multiple-exe-files-in-a-single-compilation)).

First I came across https://stackoverflow.com/questions/43258609/pyinstaller-how-to-create-multiple-programs-in-one-folder, which led me to https://github.com/pyinstaller/pyinstaller/issues/1527, but the `MERGE` feature doesn't seem to be *quite* what I want because the second exe might need to unpack the first?

Eventually I found https://www.zacoding.com/en/post/pyinstaller-create-multiple-executables/, which suggests manually editing the spec file to have multiple `Analysis`, `PYZ` and `EXE` objects, all passed into a single `COLLECT()` (also hinted at [here](https://stackoverflow.com/questions/8888813/packaging-multiple-scripts-in-pyinstaller)). This actually seems to work!

Is this supported, and should it be documented? It seems definitely useful to me, in my case particularly as I wanted both a GUI and console entrypoint.

**Comment by rokm** [comment]
Yes, having multiple `Analysis`/`PYZ`/`EXE` (parametrized to produce a `onedir` build) and a shared `COLLECT()` is the best way to generate a multi-executable bundle with shared external resources (DLLs and data files). The only drawback is that pure-python modules are still duplicated across executables (each having its own embedded PYZ archive), but the only way around that is to also use `noarchive=True` to avoid using PYZ archive at all.

It is supported. It is probably undocumented. And you need to create the spec file yourself (well, you can create a onedir spec file for each executable, and then manually merge them together).
